pinot-core/src/main/java/org/apache/pinot/core/data/recordtransformer/PinotDataType.java

```diff
@@ -632,4 +713,22 @@ public static PinotDataType getPinotDataType(FieldSpec fieldSpec) {
             "Unsupported data type: " + dataType + " in field: " + fieldSpec.getName());
     }
   }
+
+  public static PinotDataType getPinotDataTypeFromHashMap(Map<Object, Object> map) {
```

Comment:
Suggest calling it getPrimitivePinotDataTypeFromMap

HashMap is an implementation detail so let's not use that. Map is better since it is also an interface and suitable to represent an abstract complex type